src/concurrency/sync.rs

```diff
@@ -392,7 +392,12 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     /// Put the thread into the queue waiting for the mutex.
     /// Once the Mutex becomes available, `retval` will be written to `dest`.
```

Comment:
Seems like the comment needs updating to explain what happens when `retval` is `None`. Seems like `dest` is entirely ignored then? Maybe make them optional together in that case?

Reply:
Done.